checks.c

```diff
@@ -1348,6 +1348,11 @@ static void CheckSerial(X509 *x509)
 static void CheckPublicKey(X509 *x509, struct tm tm_after)
 {
 	EVP_PKEY *pkey = X509_get_pubkey(x509);
+	if (pkey == NULL)
+	{
+		SetError(ERR_INVALID);
```

Comment:
I guess we should create a different error for that, and maybe even just a warning.